sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/EncryptColumnsTransformer.scala

```diff
@@ -56,22 +56,39 @@ trait EncryptDecrypt {
     new SecretKeySpec(keyBytes, "AES")
   }
 
-  def encrypt(message: String): String = {
+  def encryptECB(message: String): String = {
+    val cipher: Cipher = Cipher.getInstance(ECB_ALGORITHM_STRING)
+    val secretKey = generateAesKey(keyAsBytes)
+    cipher.init(Cipher.ENCRYPT_MODE, secretKey)
```

Comment:
For performance reasons i suggest to initialize the cipher outside this function, otherwise this is done for every row.

Reply:
Can be kind of agree for performance reason. But then, according to the [documentation ](https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/javax/crypto/Cipher.html#doFinal()), you need to handle any exceptions of the method doFinal. Otherwise it will make crash your implementation.